src/vaadin-grid-column.html

```diff
@@ -470,9 +470,30 @@
         this.parentElement._columnPropChanged('hidden', hidden);
       }
 
-      this._allCells.forEach(cell => this._toggleAttribute('hidden', hidden, cell));
-
       if (!!hidden !== !!this._previousHidden && this._grid) {
+        if (this._previousHidden && this._cells) {
+          this._cells.forEach(cell => {
+            this._grid.appendChild(cell);
```

Comment:
Cells (´<td>´ elements) should never appear as direct children of `<vaadin-grid>`. This loop should probably be removed

Reply:
When I remove this loop, an error is raised at the `__setColumnTemplateOrRenderer` method for the same class, when it calls `this._grid.__getRowModel(cell.parentElement)` and `parentElement` is `null`.  Not sure if I could first test for the `parentElement` and skip.

Reply:
It gets invoked after every row update (even if not yet all column cells are added on rows). You could filter out only the cells that have been attached with `cells.filter(cell => cell.parentElement).forEach(cell => {...` or better yet not loop the cells until all of them are attached (more performant)

Reply:
How could we hold the loop until all cells are attached? 🤔

Reply:
Moved the logic of handling cells from a column that is moving from `hidden` to visible, so that `_renderColumnTree` of `grid` only notifies the change of cells after the last cell is reattached to the proper parent.